in-progress/12487-prd-fast-sync.md

```diff
@@ -0,0 +1,57 @@
+# Fast-Sync Project Requirements Document
+
+- Owners: @spalladino, @aminsammara
+- Approvers:
+  - @AndreOxski
+  - @charlielye
+  - @PhilWindle
+  - @joshcrites
+
+## Background
+
+Syncing an L2 node requires crawling L1 since L2 genesis, downloading events and blobs, and updating world-state for every update. This is time-consuming, and proportional to the length of the chain.
+
+## Desired User Flow
+
+Node operators should be able to sync the chain in nearly constant time by downloading a recent snapshot, and syncing only the most recent blocks. Operators should be able to do this by specifying a snapshot URL to sync from.
+
+This flow is available in most Ethereum clients.
+
+### Node operators
+
+Users SHOULD be able to specify which method to use to sync their clients. Methods are either `fast` (snapshots) or `full` (for syncing from the chain). Default is `fast`. For example:
+`aztec start --node --publisher --sync-method fast`
+
+Users SHOULD also be able to specify the location for the snapshot. Protocols supported are ipfs and https. For example:
+`aztec start --node --publisher --sync-method fast --snapshot-url ipfs://...`
+
+Users SHOULD also be able to specify the location for a snapshot index, which is a machine-readable file that lists the latest snapshots. Protocols supported https only. For example:
+`aztec start --node --publisher --sync-method fast --snapshot-index https://...`
+
+### Labs
+
+Snapshots for external networks must be generated and uploaded by Labs to IPFS on at most a 2-week basis, ideally weekly. This process should be automated.
+
+## Requirements
+
+- Snapshots can be client-specific.
```

Comment:
Please could you give an example of what a snapshot should / must contain? Surely there are some things that a snapshot _must_ contain?
(Sorry for sending these comments as lots of individual comments - I should've started a review, to reduce noise for you. Each idea came to me one-by-one, after I'd read the whole thing).

Reply:
Added in a new commit:

> The snapshot should contain all information needed for the client to be considered synced up to a given L2 chain tip. For the current client being built, this is the archiver and world state databases. Snapshot should not include any data whatsoever from the P2P layer. This should only be data syncable from L1. The resulting state of downloading a snapshot should be indistinguishable from a full sync from L1.